recipes/newmat/all/conanfile.py

```diff
@@ -0,0 +1,84 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, copy, get, load, collect_libs, rmdir
+import os
+
+required_conan_version = ">=1.53.0"
+
+
+class NewmatConan(ConanFile):
+    name = "newmat"
+    package_type = "library"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://newmat.net"
+    description = "Manipulate a variety of types of matrices using standard matrix operations."
+    settings = "os", "compiler", "build_type", "arch"
+    topics = ("newmat", "matrix")
+    license = "MIT"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def export_sources(self):
+        copy(self, "CMakeLists.txt", self.recipe_folder, self.export_sources_folder)
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate_build(self):
+        if self.settings.os == "Windows" and self.options.shared == True:
+            raise ConanInvalidConfiguration("Not working yet. Feel free to submit a fix in conan-center")
```

Comment:
It's not supported by the project or just a recipe improvement missing? In case the project does not support it, please, update the message according, otherwise, people will provide patches for something not tested by the upstream.

Reply:
I think, the message is quite clear, that the recipe improvement missing, otherwise, it would be "submit fix upstream".

Reply:
the way it is written, it implies that the project supports it, but that the recipe is not prepared. Are you sure the project supports it too?

Reply:
I didn't do a research for this particular configuration (shared lib + Windows)
I just get a linker error. I am going to mark this configuration as invalid now and fix this issue later.

Reply:
@uilianries it doesn't imply project supports it, how did you assume that? it just say configuration doesn't work. why exactly - requires an additional research. it might be a bug in recipe, or in conan, or in project, or in compiler, etc. this will require a significant time to figure it out on its own.